prow/config/config.go

```diff
@@ -262,18 +262,21 @@ type Branding struct {
 
 // Load loads and parses the config at path.
 func Load(prowConfig, jobConfig string) (*Config, error) {
-	nc, err := loadHelper(prowConfig, jobConfig)
+	nc, err := loadConfig(prowConfig, jobConfig)
 	if err != nil {
 		return nil, err
 	}
-
-	if err := nc.finalizeDecoration(); err != nil {
+	if err := nc.finalizeJobConfig(); err != nil {
+		return nil, err
+	}
+	if err := nc.validateJobConfig(); err != nil {
 		return nil, err
 	}
 	return nc, nil
 }
 
-func loadHelper(prowConfig, jobConfig string) (*Config, error) {
+// loadConfig loads one or multiple config files and returns a config object.
+func loadConfig(prowConfig, jobConfig string) (*Config, error) {
```

Comment:
I feel it would be easier to just have a loadProwConfig and loadJobConfig and change yamlToConfig like so:
```golang
func yamlToConfig(path string, config interface{}) error {}
```
That way you know what you are getting, and you can separate the parsing.

Reply:
yeah I'll need to do another pass once everything is migrated, currently the main config has both prow/job configs which means we need to call twice against the main config?